Failure in test {yaml=painless/30_search/Exception on negative score}

This test failed in:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=debian/79/console


The error is: 

```
java.lang.AssertionError: Failure at [painless/30_search:463]: expected [400] status code but api [search] returned [200 OK] [{"took":39,"timed_out":false,"_shards":{"total":2,"successful":1,"skipped":0,"failed":1,"failures":[{"shard":0,"index":"test","node":"NFi9eDrdQ5WQvhylFyC61w","reason":{"type":"illegal_argument_exception","reason":"script score function must not produce negative scores, but got: [-9.0]"}}]},"hits":{"total":0,"max_score":null,"hits":[]}}]
	at __randomizedtesting.SeedInfo.seed([A614BD8726AD3F08:2E40825D885152F0]:0)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:407)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:384)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

The REPRO command is:

```
REPRODUCE WITH: ./gradlew :modules:lang-painless:integTestRunner \
  -Dtests.seed=A614BD8726AD3F08 \
  -Dtests.class=org.elasticsearch.painless.LangPainlessClientYamlTestSuiteIT \
  -Dtests.method="test {yaml=painless/30_search/Exception on negative score}" \
  -Dtests.security.manager=true \
  -Dtests.locale=mk-MK \
  -Dtests.timezone=America/Campo_Grande \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

reproduces locally.

--- a/modules/lang-painless/src/test/resources/rest-api-spec/test/painless/30_search.yml
+++ b/modules/lang-painless/src/test/resources/rest-api-spec/test/painless/30_search.yml
@@ -464,6 +464,7 @@
         catch: bad_request
         search:
             index: test
+            allow_partial_search_results: false
             body:
                 query:
                     function_score:

